src/Core/src/Platform/Android/ImageButtonExtensions.cs

```diff
@@ -41,13 +42,12 @@ public static void UpdateCornerRadius(this ShapeableImageView platformButton, IB
 				.Build();
 		}
 
-		public static void UpdatePadding(this ShapeableImageView platformButton, IImageButton imageButton)
+		public static async void UpdatePadding(this ShapeableImageView platformButton, IImageButton imageButton)
 		{
 			platformButton.SetContentPadding(imageButton);
-			platformButton.Post(() =>
-			{
-				platformButton.SetContentPadding(imageButton);
-			});
+
+			// see: https://github.com/material-components/material-components-android/issues/2063
+			await Task.Yield();
 			platformButton.SetContentPadding(imageButton);
 		}
```

Comment:
@jsuarezruiz this code was causing a leak, see details on the commit message: d1d2ca0c0dabc8e43d1b9db67698fdd8529e3a82

Will a test fail, if the problem in 1bbe79de61f241217f88207b1272179ff66e6733 comes back?

Reply:
Yes, there is a specific test to validate the padding: `PaddingInitializesCorrectly`